rasa/core/processor.py

```diff
@@ -290,6 +290,30 @@ def _log_slots(tracker):
         if slot_values.strip():
             logger.debug("Current slot values: \n{}".format(slot_values))
 
+    def _log_unseen_intent(self, parse_data: Dict[Text, Any]) -> None:
+        """check if the NLU picks up intent that aren't in the domain.
+        """
+        intent = parse_data["intent"]["name"]
+        if not intent or intent == "":
+            return
+        if intent not in self.domain.intents and self.domain:
+            logger.warning(
+                "Interpreter parsed an intent '{}' "
+                "that is not defined in the domain.".format(intent)
+            )
+
+    def _log_unseen_enitites(self, parse_data: Dict[Text, Any]) -> None:
+        """check if the NLU picks up entities that aren't in the domain.
+        """
+        entities = parse_data["entities"]
+        for element in entities:
+            entity = element["entity"]
+            if entity not in self.domain.entities and self.domain:
```

Comment:
```suggestion
            if entity and entity not in self.domain.entities and self.domain:
```

Reply:
thanks again @tabergma , have pushed the changes and hopefully it passes the test!